### PR TITLE
Graphics/cpt 27 support deferred rendering

### PR DESCRIPTION
**Pull request by benblack90**
Apologies for it being a bit of a monster PR here - this is the full suite of support for deferred rendering. 

Alex, we will have to discuss adding directional lights, because I changed the constructor a little. This is why that section of your JSON reader is commented out.

### CODE REVIEW

**Review by Nijelous — APPROVED** [review]
Looks good, we'll discuss changing the JSON Parser tomorrow and I'll get that all implemented.

**Review by squiree98 — APPROVED** [review]

File: CSC8503/TutorialGame.cpp [hunk 1]
```diff
@@ -397,7 +397,7 @@ GameObject* TutorialGame::AddSphereToWorld(const Vector3& position, float radius
 		.SetScale(sphereSize)
 		.SetPosition(position);
 
-	sphere->SetRenderObject(new RenderObject(&sphere->GetTransform(), sphereMesh, basicTex, nullptr, basicShader,radius));
+	sphere->SetRenderObject(new RenderObject(&sphere->GetTransform(), sphereMesh, mFloorAlbedo, mFloorNormal, basicShader,radius));
```
Comment (erendgrmnc) [inline CSC8503/TutorialGame.cpp:400]: Incase something prevents "RenderObject" to be created, compiler throws an error in the accurate line when we seperate object creations like this. In other words; it's easier to debug 🕺 

auto* renderObject = new RenderObject(&sphere->GetTransform();
sphere->SetRenderObject(renderObject , sphereMesh, mFloorAlbedo, mFloorNormal, basicShader,radius));

File: CSC8503CoreClasses/GuardObject.cpp [hunk 2]
```diff
@@ -89,6 +89,8 @@ BehaviourAction* GuardObject::Patrol() {
 		}
 		else if (state == Ongoing) {
 			if (mCanSeePlayer == false) {
+				//std::cout << "Lost em";
```
Comment (erendgrmnc) [inline CSC8503CoreClasses/GuardObject.cpp:92]: Maybe we can add a todo or another comment line to describe what why this line is commented out rather then deleting it ✌️

File: CSC8503CoreClasses/JsonParser.cpp [hunk 3]
```diff
@@ -154,13 +154,13 @@ void JsonParser::WriteVariable(std::vector<std::map<std::string, float>>& keyVal
 		}
 		break;
 
-	case DirectionalLight:
+	/*case DirectionalLight:
```
Comment (erendgrmnc) [inline CSC8503CoreClasses/JsonParser.cpp:157]: Will be great to add a todo comment ✌️

**Review by erendgrmnc — COMMENTED** [review]
Thanks for your good work🥇 🦝

**Pull request merged**